docs/pbc/all/search/202410.0/base-shop/third-party-integrations/visual-search-powered-by-openai/install-visual-search-powered-by-openai.md

```diff
@@ -0,0 +1,149 @@
+search.with.your-images.error,Etwas ist schiefgelaufen. Bitte versuchen Sie es mit einem anderen Bild,de_DE
+search.with.your-images.button,Upload an image,en_US
+search.with.your-images.button,Bild hochladen,de_DE
+search.with.your.images.photo.button,Take a photo,en_US
+search.with.your.images.photo.button,Foto aufnehmen,de_DE
+search.with.your-images.no-results,"No results found. Please try another image.",en_US
+search.with.your-images.no-results,"Keine Ergebnisse gefunden. Bitte versuchen Sie es mit einem anderen Bild.",de_DE
+```
+
+### 3) Enable controllers
+
+Register the following route providers on the Storefront:
+
+| PROVIDER                         | NAMESPACE                                   |
+|----------------------------------|---------------------------------------------|
+| ImageSearchAiRouteProviderPlugin | SprykerEco\Yves\ImageSearchAi\Plugin\Router |
+
+**src/Pyz/Yves/Router/RouterDependencyProvider.php**
+
+```php
+<?php
+
+namespace Pyz\Yves\Router;
+
+use Spryker\Yves\Router\RouterDependencyProvider as SprykerRouterDependencyProvider;
+use SprykerEco\Yves\ImageSearchAi\Plugin\Router\ImageSearchAiRouteProviderPlugin;
+
+class RouterDependencyProvider extends SprykerRouterDependencyProvider
+{
+    /**
+     * @return array<\Spryker\Yves\RouterExtension\Dependency\Plugin\RouteProviderPluginInterface>
+     */
+    protected function getRouteProvider(): array
+    {
+        return [
+            new ImageSearchAiRouteProviderPlugin(),
+        ];
+    }
+}
+```
+
+{% info_block warningBox "Verification" %}
+
+To validate the route provider send an empty POST request to https://mysprykershop.com/search-ai/image, you should obtain the `{"success": false}` JSON response with 400 code.
+
+{% endinfo_block %}
+
+### 4) Set up widgets
+
+1. To enable widgets, register the following plugins:
+
+| PLUGIN              | SPECIFICATION                      | PREREQUISITES | NAMESPACE                            |
+|---------------------|------------------------------------|---------------|--------------------------------------|
+| ImageSearchAiWidget | Displays the image uploader popup. |               | SprykerEco\Yves\ImageSearchAi\Widget |
+
+**src/Pyz/Yves/ShopApplication/ShopApplicationDependencyProvider.php**
+
+```php
+<?php
+
+namespace Pyz\Yves\ShopApplication;
+
+use SprykerEco\Yves\ImageSearchAi\Widget\ImageSearchAiWidget;
+use SprykerShop\Yves\ShopApplication\ShopApplicationDependencyProvider as SprykerShopApplicationDependencyProvider;
+
+class ShopApplicationDependencyProvider extends SprykerShopApplicationDependencyProvider
+{
+    /**
+     * @return array<string>
+     */
+    protected function getGlobalWidgets(): array
+    {
+        return [
+            ImageSearchAiWidget::class,
+        ];
+    }
+}
+```
+
+See storefront related code in the demoshops PRs as an example of how to integrate the ImageSearchAi module frontend into your project:
```

Comment:
are there additional steps to do or is it the same as described in the document?